extensions/amp-script/0.1/amp-script.js

```diff
@@ -539,6 +569,28 @@ export class AmpScriptService {
     this.cumulativeSize_ += size;
     return this.cumulativeSize_ > MAX_TOTAL_SCRIPT_SIZE;
   }
+
+  /**
+   * Calls an exported function on a specific amp-script worker and returns the result.
+   *
+   * @param {string} ampScriptId
+   * @param {string} fnIdentifier
+   * @return {Promise<*>}
+   */
+  callFunction(ampScriptId, fnIdentifier) {
```

Comment:
Why have this intermediary function on `AmpScriptService` at all? 

There's only a single caller but if you want to avoid the boilerplate you could also hook into the internal actions API e.g. `registerAction`.

Reply:
This would actually solve a lot of interesting problems around args parsing so I'd love to make it work.
I'm not seeing a straightforward way to pipe through the return value though

Reply:
Ah yea, I guess `executeAction` could work but not sure it's a net win actually.

Reply:
Yeah. Calling `executeAction` would mean I'd still need to locate the correct element, so wouldn't help much here. A nice aspect is that if I registered the action, we'd get part (1) of this issue implemented for free: https://github.com/ampproject/amphtml/issues/24763

Also, I could imagine us moving in a direction eventually where devs pass a mixture of literal arguments/ amp-state arguments into the function. I skirt the issue by banning it entirely, but if we utilized the actions parser it could be easier to implement.  This would become especially important if we wanted to make amp-script functions accessible from bind expressions.

Interop with bind expressions would let devs write a bound src with amp-state vars and have it automatically updated when the state variable is updated. 
```html
<amp-list [src]="amp-script:fns.fetchListData(filterArgs)">
```

The current workaround requires chaining a `list.refresh()` action after the `setState` action.